app/extensions/order-discount-zig/src/main.zig

```diff
@@ -76,38 +52,24 @@ pub fn main() u8 {
 
     const allocator = arena.allocator();
 
-    // Read all stdin to an ArrayList
-    var inputStr = std.ArrayList(u8).init(allocator);
-    var stdin = std.io.getStdIn().reader();
-    var buffer: [1024]u8 = undefined;
-    while (true) {
-        const bytesRead = stdin.read(&buffer) catch |err| {
-            std.debug.print("Error reading function input: {!}",.{err});
-            return 1;
-        };
-        if (bytesRead == 0) {
-            break; // No more data
-        }
-        inputStr.appendSlice(buffer[0..bytesRead]) catch |err| {
-            std.debug.print("Error reading function input: {!}",.{err});
-            return 1;
-        };
-    }
+    var buffer: [3 * 1024]u8 = undefined;
```

Comment:
Little trick as our input is limited to max 3K afaik

Reply:
Input actually isn't limited today, but 3K is probably still a better default.

Reply:
An actually, I think this is causing zig to fail on the large cart, thus the abnormally slow instruction count there.